src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2247,6 +2247,42 @@ internal void AddModuleInitializerMethod(MethodSymbol method)
             LazyInitializer.EnsureInitialized(ref _moduleInitializerMethods).Add(method);
         }
 
+        private ConcurrentSet<(InterceptsLocationAttributeData, MethodSymbol)>? _interceptions;
+
+        internal void AddInterception(InterceptsLocationAttributeData location, MethodSymbol interceptor)
+        {
+            Debug.Assert(!_declarationDiagnosticsFrozen);
+            LazyInitializer.EnsureInitialized(ref _interceptions).Add((location, interceptor));
+        }
+
+        internal (InterceptsLocationAttributeData data, MethodSymbol interceptor)? GetInterceptor(Location? callLocation)
+        {
+            if (_interceptions is null || callLocation is null)
+            {
+                return null;
+            }
+
+            foreach (var (interceptsLocation, interceptor) in _interceptions)
+            {
+                var callLineColumn = callLocation.GetLineSpan().Span.Start;
+                if (interceptsLocation.FilePath == callLocation.SourceTree!.FilePath
```

Comment:
>! [](http://example.com/codeflow?start=74&length=1)

We should use an assertion instead of a suppression since there is no earlier check in this method that implies this knowledge already.